src/eip712.rs

```diff
@@ -1223,15 +1302,73 @@ mod tests {
             },
             "proof": {
               "verificationMethod": "did:example:aaaabbbb#issuerKey-1",
-              "created": "2010-01-01T19:23:24Z",
+              "created": "2021-07-09T19:47:41Z",
               "proofPurpose": "assertionMethod",
               "type": "EthereumEip712Signature2021"
             }
           }
         });
         assert_eq!(
-            serde_json::to_value(typed_data).unwrap(),
+            serde_json::to_value(&typed_data).unwrap(),
             expected_typed_data
         );
+
+        let jwk: crate::jwk::JWK = serde_json::from_value(json!({
+            "kty": "EC",
+            "crv": "secp256k1",
+            "x": "cmbYyDC6cbm807_OmFNYP4CLEL0aB2F1UG683SxFkXM",
+            "y": "zBw5HAh0cJM4YimSQvtYM1HFhzUXVUgrDhxJ70aajt0",
+            "d": "u7QuEl6W0XNppEY0iMVjATT99tC9acwV3Z2keEqvKGo"
+        }))
+        .unwrap();
+        eprintln!("jwk {}", serde_json::to_string(&jwk).unwrap());
+
+        let td_jcs = serde_jcs::to_string(&typed_data).unwrap();
+        // Wrap string with line breaks
+        // https://stackoverflow.com/a/57032118
+        let jcs_lines = td_jcs
+            .chars()
+            .enumerate()
+            .flat_map(|(i, c)| {
+                if i != 0 && i % 90 == 0 {
+                    Some('\n')
+                } else {
+                    None
+                }
+                .into_iter()
+                .chain(std::iter::once(c))
+            })
+            .collect::<String>();
+        eprintln!("JCS: [\n{}\n]", jcs_lines);
+
+        // Sign proof
+        use crate::ldp::ProofSuite;
+        let bytes = typed_data.bytes().unwrap();
+        let ec_params = match &jwk.params {
+            crate::jwk::Params::EC(ec) => ec,
+            _ => unreachable!(),
+        };
+        use k256::ecdsa::signature::Signer;
+        let secret_key = k256::SecretKey::try_from(ec_params).unwrap();
+        let signing_key = k256::ecdsa::SigningKey::from(secret_key);
+        let sig: k256::ecdsa::recoverable::Signature = signing_key.try_sign(&bytes).unwrap();
+        let sig_bytes = &mut sig.as_ref().to_vec();
+        // Recovery ID starts at 27 instead of 0.
+        sig_bytes[64] = sig_bytes[64] + 27;
+        let sig_hex = crate::keccak_hash::bytes_to_lowerhex(sig_bytes);
+        let mut proof = proof.clone();
+        proof.proof_value = Some(sig_hex.clone());
+        eprintln!("proof {}", serde_json::to_string(&proof).unwrap());
+
+        // Verify the VC/proof
+        let mut vc = vc.clone();
+        vc.add_proof(proof.clone());
+        vc.validate().unwrap();
+        let verification_result = vc.verify(None, &DIDExample).await;
+        println!("{:#?}", verification_result);
+        assert!(verification_result.errors.is_empty());
+
+        assert_eq!(sig_hex, "0x5fb8f18f21f54c2df8a2720d0afcee7dbbb18e4b7a22ce6e8183633d63b076d329122584db769cd78b6cd5a7094ede5ceaa43317907539187f1f0d8875f99e051b");
```

Comment:
That could be put before the verification

Reply:
When changing the test vector and/or algorithm, the signature will change, but we still want to know that the verification passes before doing anything further. So it's intended to be useful for development purposes to assert this signature equality last.

Reply:
Makes sense :+1: